### DISCUSSION TITLE
Feedback: Positive Data Acceptance

### DISCUSSION THREAD

**Discussion by Stranger6667** [schemathesis/schemathesis#2499] [Q&A]
We've introduced an experimental feature to verify that schema-conforming data receives expected success responses. This check aims to highlight mismatches between the API schema and actual behavior, potentially uncovering issues in schema definition or API implementation.

## How to Enable

CLI option: `--experimental=positive_data_acceptance`

## Configuration

You can customize the allowed status codes:

```bash
--experimental-positive-data-acceptance-allowed-statuses=200,401,403,404
```

By default, the check allows 2xx, 404, 401, and 403 status codes

## Feedback

We're eager to hear your feedback:

- Has this check helped identify discrepancies between your API schema and implementation?
- Are there scenarios where you've encountered false positives? If so, what were they?
- Is the configuration flexible enough for your use cases?
- Have you encountered any issues or unexpected behavior?
- Any additional feedback or suggestions for improvement?

Your feedback is crucial for refining this feature. Thank you for helping us enhance Schemathesis!

**Comment by uthbees** [comment]
I ran into a minor issue with this feature on the latest docker image (`schemathesis/schemathesis:4.0.0-alpha.4`). It seems that the feature is enabled by default, even though it's only supposed to be enabled with the CLI option. To make things even more confusing, the `--experimental-positive-data-acceptance-allowed-statuses` parameter is not processed unless `--experimental=positive-data-acceptance` is provided - so it's enabled by default, but you can't configure it.

Good feature though!

**Reply by Stranger6667** [reply]
Thanks for the report! My plan for alpha 6 is to cleanup all those experimental features and their configurations and provide a clear way to enable & configure them (likely via a config file)

**Comment by Anas-Salha** [comment]
Hi,

I've had an issue with this feature generating false failures.

I have schema defined as follows:
```
    PatchTodo:
      type: object
      additionalProperties: false
      properties:
        text:
          type: string
          example: Make dinner
        completed:
          type: boolean
          example: false
      minProperties: 1
```

I expect a schema-compliant request to contain at least one of the properties, hence the `minProperties: 1` field. 
However, schemathesis seems to think an empty json is a schema-compliant request. 

Failure output below:

```
1. Test Case ID: krdMX2

- API rejected schema-compliant request

    Valid data should have been accepted
    Expected: 2xx, 401, 403, 404, 5xx

[400] Bad Request:

    `{"error":"text and completed fields missing - provide at least one"}`

Reproduce with: 

    curl -X PATCH -H 'Content-Type: application/json' -d '{}' http://localhost:8080/todos/1
```

**Reply by Stranger6667** [reply]
@Anas-Salha 

Sorry for not getting back to you earlier. I've checked it manually with the latest version (4.4.4) and wasn't able to reproduce it. There were some changes regarding the "negative" / "positive" metadata, so it should be much more reliable right now - before there were some mutations that affected the payloads, and that metadata became stale.

If the issue is not resolved for you in the latest version, please let me know. I'd be happy to dig deeper
